Userland/Libraries/LibCore/DateTime.cpp

```diff
@@ -325,7 +325,7 @@ ByteString DateTime::to_byte_string(StringView format, LocalTime local_time) con
     return MUST(to_string(format, local_time)).to_byte_string();
 }
 
-Optional<DateTime> DateTime::parse(StringView format, StringView string)
+Optional<DateTime> DateTime::parse(StringView format, StringView string, unsigned* microseconds)
```

Comment:
I don't particularly like this API, but adding microseconds+milliseconds to the class also felt a bit off 🤔

Reply:
Wait, what's wrong with adding it to DateTime?

Reply:
Maybe there isn't anything wrong at all. It just felt a bit strange since it represents a time_t which doesn't have that type of resolution.

Probably just best to have a microsecond member to store this but no millisecond one thinking about it more

Reply:
I was thinking that it could maybe be a bit confusing with APIs such as: `DateTime::set_time` and `DateTime::from_timestamp` etc